eis_toolkit/vector_processing/proximity_tool.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import geopandas as gpd
+from numbers import Number
+from beartype import beartype
+from beartype.typing import Union
+from rasterio import profiles, transform
+from eis_toolkit.vector_processing.distance_computation import distance_computation
+
+@beartype
+def calculate_proximity(geodataframe: gpd.GeoDataFrame, raster_profile: Union[profiles.Profile, dict], maximum_distance: Number) -> np.ndarray:
+
+    """ Interpolates the distance values calculated by the distance_computation function between 0 and 1.
+        1 denots the value inside the polygon and 0 at the maximum distance.
+        If maximum_distance value is not provided, the program sets this value to the maximum value 
+        in the provided distance matrix.
+        Uses linear interpolation to calculate the distance from the polygon.
```

Comment:
The format of the docstring should follow that of other tools. I believe the line "If maximum_distance value is not provided, the program sets this value to the maximum value " is also outdated as `maximum_distance` is required